File: Program's_Contributed_By_Contributors/Python_Programs/LongestPalindromicSubstring.py
//Longest Palindromic Substring
//https://leetcode.com/problems/longest-palindromic-substring/

class Solution:
    def countPalindromeAroundCenter(self, lo: int, hi: int, s: str) -> str:
        while lo >= 0 and hi < len(s):
            if s[lo] != s[hi]:
                break
            lo -= 1
            hi += 1
        res = s[(lo+1):hi]
        return res
                
    def longestPalindrome(self, s: str) -> str:
        ans = ""
        for i in range (len(s)):
            s1 = self.countPalindromeAroundCenter(i, i, s)
            s2 = self.countPalindromeAroundCenter(i, i + 1, s)
            if len(s1) > len(ans):
                ans = s1
            if len(s2) > len(ans):
                ans = s2
        return ans

File: Program's_Contributed_By_Contributors/Java_Programs/LongestPalindromicSubstring.java
// https://leetcode.com/problems/longest-palindromic-substring/
package com.sharan;
// Time Limit Exceeded.
public class LongestPalindromicSubstring {
    public static String longestPalindrome(String s) {
        String ans = "";
        for(int i = 0; i < s.length(); i++){
            StringBuilder temp = new StringBuilder();
            temp.append(s.charAt(i));
            for(int j = i + 1; j < s.length(); j++){
                temp.append(s.charAt(j));
                String tempAns = temp.reverse().toString();
                temp.reverse();
                if(temp.toString().contains(tempAns)){
                    if(ans.length() < tempAns.length()){
                        ans = tempAns;
                    }
                }
            }
        }
        if(ans.length() == 0){
            return String.valueOf(s.charAt(0));
        }
        else return ans;
    }
}
